### ISSUE TITLE
in AFURLRequestSerialization ,AFQueryStringFromParameters()  need check parameters type.

### ISSUE THREAD

**Issue by 5nn** [AFNetworking/AFNetworking#4170]
When app is running, the value of parameters is string @"", and "func AFQueryStringFromParameters (parameters)"  (AFURLRequestSerialization, line 504),will continue to run, and eventually the value of query is @" =". So we need to do parameter type checking when executing " AFQueryStringFromParameters()"

**Comment by 5nn** [comment]
（line 504） Suggested modification：
 switch (self.queryStringSerializationStyle) {
                case AFHTTPRequestQueryStringDefaultStyle:
                    query = [parameters isKindOfClass:[NSDictionary class]] ?AFQueryStringFromParameters(parameters): nil;
                    break;
            }

**Comment by 5nn** [comment]
🙂 （line 504）Suggested modification：
switch (self.queryStringSerializationStyle) {
                case AFHTTPRequestQueryStringDefaultStyle:
                    query = [parameters isKindOfClass:[NSDictionary class]] ?AFQueryStringFromParameters(parameters): nil;
                    break;
            }

**Comment by ymsmile** [comment]
I encountered a similar problem. My parameter is an encrypted string. The parameters passed to the server through this framework are inconsistent with those issued by the client, so the server cannot decrypt it. I made the following configuration to enable the server to properly decrypt the parameters, create instances of **AFHTTPSessionManager**, rewrite its **requestSerializer** attribute **setQueryStringSerializationWithBlock** method, method is as follows:
`
[_sessionManager.requestSerializer setQueryStringSerializationWithBlock:^NSString * _Nonnull(NSURLRequest * _Nonnull request, id  _Nonnull parameters, NSError * _Nullable __autoreleasing * _Nullable error)  {
 return parameters;
        }];
`
